inc/modules/import/wordpress/class-wxr.php

```diff
@@ -171,6 +184,73 @@ function setCurrentImportOption( array $upload ) {
 		return update_option( 'pressbooks_current_import', $option );
 	}
 
+	/**
+	 * Insert a term with the meta associated
+	 * @param $term
+	 * @return array|int[]|mixed|\WP_Error
+	 */
+	public function insertTerm( $term ) {
+		$results = wp_insert_term(
+			$term['term_name'],
+			$term['term_taxonomy'],
+			[
+				'description' => $term['term_description'],
+				'slug' => $term['slug'],
+			]
+		);
+		if ( ! empty( $term['termmeta'] ) && is_array( $results ) ) {
+			foreach ( $term['termmeta'] as $termmeta ) {
+				$value = $termmeta['value'];
+				// Copy contributor_picture from remote server to local
+				if ( $termmeta['key'] === 'contributor_picture' && $value ) {
+					$image_id = $this->downloads->fetchAndSaveUniqueImage( $value );
+					if ( $image_id > 0 ) {
+						$value = wp_get_attachment_url( $image_id );
+					}
+				}
+				add_term_meta( $results['term_id'], $termmeta['key'], $value, true );
+			}
+		}
+
+		return $results;
+	}
+
+	/**
+	 * This function search for existent contributors by checking every field to look for exact duplicates.
+	 * @param $term
+	 * @return false|mixed
+	 */
+	public function findExistentTerm( $term ) {
+
+			$args = [
+				'taxonomy' => 'contributor',
+				'hide_empty' => false,
+				'meta_query' => [], // phpcs:ignore
+			];
+
+			$fields_to_compare = [ 'contributor_first_name', 'contributor_last_name', 'contributor_prefix', 'contributor_suffix' ];
```

Comment:
Os, a question here: does it means if we find a contributor with let's say same first name and prefix, but the rest is different, do we consider it as a existent term?

Reply:
Only If all of those fields are the same we consider that is the same term